csrc/ir_nodes.cpp

```diff
@@ -1329,6 +1329,253 @@ Val* GroupedWelfordOp::getInitValOfOutput(Val* output_val) const {
 
 NVFUSER_DEFINE_CLONE_AND_CREATE(GroupedWelfordOp)
 
+//==============================================================================================================================
+
+// MmaOp utils
+namespace MmaOpUtils {
+
+// The expected number of concrete domains for gemm
+constexpr size_t expected_gemm_cdomains = 2;
+
+// A helper structure used to gather all data created during analysis
+struct MmaOpDetails {
```

Comment:
How about merging this with `OptionsInMma`?

Reply:
I defined `MmaOpDetails` only to have a clean way of returning analysis results from `getMmaOpDetails(...)` to `MmaOp::MmaOp` so the populated data can be stored in `MmaOp` attributes. To be honest, I could have used `std::tuple<...>` instead of defining this structure. This is the reason why it is defined in a namespace in the source file, not in header.

I need to dive into `OptionsInMma` to understand where it is used before deciding to extend it. In fact, in its definition, there is a note that in the end it should contain only mma macro.

Reply:
It's fine. Let's leave this PR as is, and only worry about refactoring in the future.